app.py

```diff
@@ -0,0 +1,210 @@
+        help="Which voice preset to use.",
+        index=1,
+    )
+    with st.expander("Advanced"):
+        col1, col2 = st.columns(2)
+        with col1:
+            """#### Model parameters"""
+            candidates = st.number_input(
+                "Candidates", help="How many output candidates to produce per-voice.", value=3
+            )
+            sampler = st.radio(
+                "Sampler",
+                SAMPLERS,
+                help="override the sampler used for diffusion (default depends on preset)",
+                index=1
+            )
+            steps = st.number_input(
+                "Steps",
+                help="Override the steps used for diffusion (default depends on preset)",
+                value=100,
+            )
+            seed = st.number_input(
+                "Seed", help="Random seed which can be used to reproduce results.", value=-1
+            )
+            if seed == -1:
+                seed = None
+            """#### Directories"""
+            output_path = st.text_input(
+                "Output Path", help="Where to store outputs.", value="results/"
+            )
+            model_dir = st.text_input(
+                "Model Directory",
+                help="Where to find pretrained model checkpoints. Tortoise automatically downloads these to .models, so this"
+                "should only be specified if you have custom checkpoints.",
+                value=MODELS_DIR,
+            )
+
+
+
+        with col2:
+            """#### Optimizations"""
+            high_vram = not st.checkbox(
+                "Low VRAM",
+                help="Re-enable default offloading behaviour of tortoise",
+                value=True,
+            )
+            half = st.checkbox(
+                "Half-Precision",
+                help="Enable autocast to half precision for autoregressive model",
+                value=False,
+            )
+            kv_cache = st.checkbox(
+                "Key-Value Cache",
+                help="Enable (partially broken) kv_cache usage, leading to drastic speedups but worse memory usage + results",
+                value=False,
+            )
+            cond_free = st.checkbox(
+                "Conditioning Free", help="Force conditioning free diffusion", value=True
+            )
+            no_cond_free = st.checkbox(
+                "Force Not Conditioning Free",
+                help="Force disable conditioning free diffusion",
+                value=False,
+            )
+
+            """#### Debug"""
+            produce_debug_state = st.checkbox(
+                "Produce Debug State",
+                help="Whether or not to produce debug_state.pth, which can aid in reproducing problems. Defaults to true.",
+                value=True,
+            )
+
+    if 'tts' not in st.session_state:
+        st.session_state.tts = TextToSpeech(models_dir=model_dir, high_vram=high_vram, kv_cache=kv_cache)
+    tts = st.session_state.tts
```

Comment:
this just doesn't work. changing the checkboxes on the browser does not actually reload the configuration settings for TextToSpeech(). this is a significant issue. i'll try to fix it too

Reply:
i [patched](https://github.com/152334H/tortoise-tts-fast/pull/12/commits/bffb8ddcf47cbc0b56f91c80ed1d769ff357f31e) this to reinit the TTS object every time the relevant config options are changed. this is suboptimal and a bettter approach would be to have a specific `TextToSpeech.update()` method that does the bare minimum required work to change options.